backend/BudgetMate/data_ingestion/admin.py

```diff
@@ -1,3 +1,78 @@
 from django.contrib import admin
+from .models import (
+    BankBrand,
+    BankAccount,
+    AccountStatement,
+    StatementLine,
+    Category,
+    SubCategory,
+)
 
-# Register your models here.
+
+@admin.register(BankBrand)
+class BankBrandAdmin(admin.ModelAdmin):
+    list_display = ("name",)
+
+
+@admin.register(BankAccount)
+class BankAccountAdmin(admin.ModelAdmin):
+    model = BankAccount
+    list_display = ("account_number", "bank", "user", "description")
+
+
+@admin.register(AccountStatement)
+class AccountStatementAdmin(admin.ModelAdmin):
+    model = AccountStatement
+    list_display = (
+        "statement_type",
+        "start_date",
+        "end_date",
+        "bank_account_number",
+        "bank_name_acc",
+        "user_account",
+    )
+
+    def bank_account_number(self, obj):
+        return obj.bank_account.account_number
+
+    def bank_name_acc(self, obj):
+        return obj.bank_account.bank.name
+
+    def user_account(self, obj):
+        return obj.bank_account.user
+
+
+@admin.register(StatementLine)
+class StatementLineAdmin(admin.ModelAdmin):
+    model = StatementLine
+    list_display = (
+        "account_statement",
+        "operation_type",
+        "amount",
+        "operation_date",
+        "user_account",
+        "category",
+        "sub_category",
+    )
+
+    def user_account(self, obj):
+        return obj.account_statement.bank_account.user
+
+    list_filter = (
+        "account_statement",
+        "operation_type",
+        "amount",
+        "operation_date",
+        "category",
+        "sub_category",
+    )
```

Comment:
You certainly don'"t want an item in the filter list per value : 

screenshot from docs : 
![image](https://user-images.githubusercontent.com/70256364/236686274-9af5b53c-45ac-4179-beaf-0dab2f5cbb72.png)

Doing like you did, you will have one filter on the right _per day_ meaning you will end up having hundreds or thousands lines in the filter category for dates alone ! And also one for each amount. 

I believe that what you are after are [search_fields](https://docs.djangoproject.com/en/4.2/ref/contrib/admin/#django.contrib.admin.ModelAdmin.search_fields) :slightly_smiling_face:

Reply:
Replaced by search_fields !